### PR TITLE
[rs] 09: Flatten crate structure of amethyst_renderer

### PR DESCRIPTION
**Pull request by torkleyy**
Partially addresses #396

### CODE REVIEW

File: src/prelude.rs [hunk 1]
```diff
@@ -3,5 +3,5 @@
 pub use app::{Application, ApplicationBuilder};
 pub use config::Config;
 pub use engine::Engine;
-pub use renderer::input::*;
+//pub use renderer::input::*;
```
Comment (torkleyy) [inline src/prelude.rs]: I'm not yet sure what we should do with this prelude module, let's decide once [rs] PRs are merged

File: amethyst_renderer/src/prelude.rs [hunk 2]
```diff
@@ -1,19 +0,0 @@
-//! Contains common types that can be glob-imported (`*`) for convenience.
```
Comment (Rhuagh) [inline amethyst_renderer/src/prelude.rs]: Convention seem to be the prelude containing traits,  i.e. things that are used implicitly, but not directly instantiated.

File: src/prelude.rs (hunk 1 above)
Comment (Rhuagh) [reply src/prelude.rs]: See above

File: amethyst_renderer/src/prelude.rs (hunk 2 above)
Comment (torkleyy) [reply amethyst_renderer/src/prelude.rs]: I heard that, yeah, but I think that many libraries have different preludes with different usages.

File: amethyst_renderer/src/config.rs [hunk 3]
```diff
@@ -4,7 +4,7 @@ use winit::WindowBuilder;
 
 /// Structure for holding the renderer configuration.
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
-pub struct Config {
+pub struct DisplayConfig {
```
Comment (Xaeroxe) [inline amethyst_renderer/src/config.rs]: Thanks for renaming this!

**Pull request merged**